### ISSUE TITLE
Set --config for htmlhint

### ISSUE THREAD

**Issue by lukelarsen** [w0rp/ale#1038]
I'm trying to get htmlhint working with my htmlhint config file. I have it created and named `.htmlhintrc` and it is in the root of my project. While htmlhint is working in Ale it seem to be using the default config for the lint rules. No matter what I change in my .htmlhintrc file it doesn't take. According to https://github.com/yaniswang/HTMLHint/wiki/Usage .htmlhintrc is the correct name and it should be in the root dir. Running htmlhint from the command line works just fine.

Doing this approach works great with stylelint. I was hoping the same would be for htmlhint. Is this a real issue, is it something wrong with htmlhint, or is it just me doing something dumb? Thanks

Current Filetype: html
Available Linters: ['htmlhint', 'proselint', 'tidy']
  Enabled Linters: ['htmlhint']
 Linter Variables:

let g:ale_html_htmlhint_executable = 'htmlhint'
let g:ale_html_htmlhint_options = '--format=unix'
let g:ale_html_htmlhint_use_global = 0
 Global Variables:

let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%s'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'always'
let g:ale_linter_aliases = {}
let g:ale_linters = {'html': ['htmlhint'], 'javascript': ['eslint'], 'css': ['stylelint']}
let g:ale_open_list = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 1
let g:ale_set_signs = 1
let g:ale_sign_column_always = 1
let g:ale_sign_error = '>>'
let g:ale_sign_offset = 1000000
let g:ale_sign_warning = '--'
let g:ale_statusline_format = ['%d error(s)', '%d warning(s)', 'OK']
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - success) htmlhint
(started) ['/bin/zsh', '-c', 'htmlhint --format=unix ''/var/folders/7m/5rzsq7915692mlnd1p2m4ny40000gn/T/nvimLAOmxp/4/acuity.html''']
(finished - exit code 1) ['/bin/zsh', '-c', 'htmlhint --format=unix ''/var/folders/7m/5rzsq7915692mlnd1p2m4ny40000gn/T/nvimLAOmxp/7/acuity.html''']

<<<OUTPUT STARTS>>>
/var/folders/7m/5rzsq7915692mlnd1p2m4ny40000gn/T/nvimLAOmxp/7/acuity.html:13:45: The attribute name of [ viewBox ] must be in lowercase. [error/attr-lowercase]

^[[31m1 problems^[[39m
<<<OUTPUT ENDS>>>

**Comment by w0rp** [comment]
The linter will need to be modified to set `--config` with the filename manually, so it can both check a temporary file and use whatever configuration file.

**Comment by w0rp** [comment]
See the `jshint` linter for an example. Several other linters do that.

**Comment by lukelarsen** [comment]
Thanks for the response and for your work on Ale. 

I'm confused about next steps. Is this something I can configure in my init.vim where I have all my vim configuration or is there work that needs to be done on Ale to get this going?

**Comment by lukelarsen** [comment]
I did try putting this in my init.vim

`let g:ale_html_htmlhint_options = '--config=~/.htmlhintrc'`

with no luck.

**Comment by w0rp** [comment]
Nah, someone needs to modify the VimL code so it automatically sets the configuration file. That's the most important thing.

**Comment by lukelarsen** [comment]
Good to know. I wish I had more knowledge about this. I have no idea where to start or I'd give it a try. I'm a html/css guy and get lots with this complicated stuff.

**Comment by w0rp** [comment]
This should work now. I made it check the temporary file and also detect and use the configuration file automatically. If you have set the configuration file path yourself with the options variable, that will be used instead.

I also removed `--format=unix` from the default options, as that format needs to always be used, or ALE just won't work. If you have that in your options variable, it will automatically be removed, so your configuration should continue to work.
